repository_service_tuf_worker/repository.py

```diff
@@ -1421,97 +1420,84 @@ def sign_metadata(
     ) -> Dict[str, Any]:
         """Add signature to metadata for pending signing event.
 
-        Add signature (from payload) to cached role metadata (from settings)
-        for the role that matches the passed rolename (from payload), if a
-        signing event exists for that role, and the signature is valid.
-
-        If the signature threshold is reached, the signing event is finalized.
+        Add signature (from payload) to cached root metadata (from settings), if
+        a signing event exists, and the signature is valid.
 
-        ** Signing event types (and details) **
+        Signing event types are 'bootstrap' or 'metadata update'.
 
-        BOOTSTRAP: Only root metadata can be updated in this event. To verify
-        the passed signature, the keys and threshold are read from the root
-        metadata to be updated itself. If the threshold is reached, the
-        bootstrap process is finalized.
-
-        METADATA UPDATE: Root, targets or delegated targets metadata can be
-        updated in this event. Depending on the metadata type, the authorized
-        public keys and threshold are read from different delegating metadata.
+        If the signature threshold is reached, the signing event is finalized,
+        otherwise it remains in pending state.
         """
-        rolename = payload["role"]
-        signature_dict = payload["signature"]
 
-        # Assert pending signing event
-        metadata_dict = self._settings.get_fresh(f"{rolename.upper()}_SIGNING")
+        def _result(status, error=None, bootstrap=None, update=None):
+            details = {}
+            if status:
+                details["message"] = "Signature Processed"
+            else:
+                details["message"] = "Signature Failed"
+            if error:
+                details["error"] = error
+            if bootstrap:
+                details["bootstrap"] = bootstrap
+            if update:
+                details["update"] = update
```

Comment:
We call the task "Metadata Update", so "update" or "metadata_update" is ok IMO.
I thought in the future to identify as "Root" or "Targets", but it doesn't give much as it could be a delegated target as well.
Maybe the role type, role name, and version? Because after the request, the user has only the task id, and retrieving this task results with more details could be helpful. WDYT?

Reply:
> Maybe the role type, role name, and version? Because after the request, the user has only the task id, and retrieving this task results with more details could be helpful. WDYT?

role name and version should be good. Should I add this here, or in a follow-up?

Reply:
btw, I think is a "good first issue" too